src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -96,6 +109,16 @@ public int stem(char s[], int len) {
     if (len < 3 || stopwords.contains(s, 0, len))
       return len;
 
+    // a String representation of the term's char[]
+    String termStr = String.valueOf(s);
+
+    //handle step 0 and step 1 exceptions
+    char[] exceptional = stepZeroExceptions.get(termStr);
+
+    if(exceptional != null) {
+      return handleException(s, exceptional);
```

Comment:
I would do the System.arraycopy here and return the exceptional.length. If you ever need to do more processing here, you can then choose to extract it in a method. This one will improve code readability (this class has more than  1k lines, so you can benefit by not adding unecessary methods).